Insert proper line returns for multiple keys

drewgates opened the pull request:
Fixed an issue where users with multiple keys in their Github account would have all keys pasted on one line in authorized_keys, separated by a single space.

Adding this line replaces that space with a newline character for each subsequent key. This does not appear to interfere with key removal.

Aranaur committed:
Insert proper line returns for multiple keys

Fixed an issue where users with multiple keys in their Github account would have all keys pasted on one line in authorized_keys, separated by a single space.

Adding this line replaces that space with a newline character for each subsequent key. This does not appear to interfere with key removal.

flplv commented on ssh-allow-friend:
```diff
@@ -160,6 +160,7 @@ setup () {
 
         mkdir -p $HOME/.ssh/
         echo $USERKEY >> $HOME/.ssh/authorized_keys
+        sed -i 's/ ssh/\nssh/g' $HOME/.ssh/authorized_keys
```
1) I am unsure which service is posting keys in the same line, as I tested [here](https://github.com/chadkouse.keys) it seems to be multiline, can you show us a example case? 
2) Can we add support to other keys type? Accordingly to http://man.he.net/man5/authorized_keys.

I'm glad for your help.
Thanks.

Aranaur committed:
Fixed to allow ssh- and ecdsa-

This should allow for ssh-rsa, ssh-dss, ecdsa-sha2-nistp256, ecdsa-sha2-nistp384, and ecdsa-sha2-nistp521 key types.

drewgates commented:
I made a change and added another line so that the rsa, dss, and ecdsa keys are supported.

As you mentioned, the multiple keys appear correctly in the web browser, but if you check the authorized_keys file while the script is running (see attached image), the newline is removed.


<img width="1254" alt="screen shot 2017-02-03 at 4 57 00 pm" src="https://cloud.githubusercontent.com/assets/7493966/22610106/d879ebe0-ea31-11e6-84a2-8154df268f23.png">

drewgates commented:
Highlighted the start of each key in the attached.

![screen shot 2017-02-03 at 4 57 00 pm2](https://cloud.githubusercontent.com/assets/7493966/22610239/837abda8-ea32-11e6-9cd9-ba9d5e29c6e2.png)

flplv commented:
True.. just confirmed the behavior here :)
Thanks for the clarification.

flplv commented:
had to revert since It will be better to fix the "echo" instead adding post processing on the auth_keys files.

aranaur commented:
That makes sense. I wasn't sure how to fix the echo. Do you have a fix or would you like me to look further into that?

flplv commented:
Just did it :)
https://github.com/flplv/ssh-allow-friend/commit/aa59f367d570b9a867f257e3e09f3a45d8a5705d
Anyways, I'm very glad that you helped.


On Fri, Feb 3, 2017 at 8:19 PM, Drew Gates <notifications@github.com> wrote:

> That makes sense. I wasn't sure how to fix the echo. Do you have a fix or
> would you like me to look further into that?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/flplv/ssh-allow-friend/pull/4#issuecomment-277378091>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEsiO7DnVj7uUL1eq4Qs8vp_k_E72V0iks5rY6fmgaJpZM4L2zb0>
> .
>

drewgates commented:
Works for me! Glad I could point out the issue.
